File: _src/05/Chapter5.CSharp/Models/Person.cs
using System;
using System.Collections.Generic;

namespace Chapter5.CSharp.Models
{
    public class Person
    {
        public Person()
        {
            Phones = new HashSet<Phone>();
            Companies = new HashSet<Company>();
            Address = new Address();
        }

        public int PersonId { get; set; }
        public int? PersonTypeId { get; set; }
        public virtual PersonType PersonType { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string MiddleName { get; set; }
        public string FullName
        {
            get
            {
                return string.Format("{0} {1}", FirstName, LastName);

            }
            set
            {
                var names = value.Split(new string[] { " " }, StringSplitOptions.RemoveEmptyEntries);
                FirstName = names[0];
                LastName = names[1];
            }
        }
        public DateTime? BirthDate { get; set; }
        public decimal HeightInFeet { get; set; }
        public bool IsActive { get; set; }
        public PersonState PersonState { get; set; }
        public Address Address { get; set; }
        public byte[] Photo { get; set; }
        public byte[] FamilyPicture { get; set; }
        public virtual ICollection<Phone> Phones { get; set; }
        public virtual ICollection<Company> Companies { get; set; }

    }

}

File: _src/05/Chapter5.VB/Models/Person.vb
Namespace Models

    Public Class Person
        Public Sub New()
            Phones = New HashSet(Of Phone)
            Companies = New HashSet(Of Company)
            Address = New Address()
        End Sub

        Property PersonId() As Integer
        Property PersonTypeId() As Integer?
        Overridable Property PersonType() As PersonType
        Property FirstName() As String
        Property LastName() As String
        Property MiddleName() As String
        Public Property FullName() As String
            Get

                Return String.Format("{0} {1}", FirstName, LastName)
            End Get
            Set(value As String)
                Dim names = value.Split(New String() {" "}, StringSplitOptions.RemoveEmptyEntries)
                FirstName = names(0)
                LastName = names(1)
            End Set
        End Property
        Property BirthDate() As DateTime?
        Property HeightInFeet() As Decimal
        Property IsActive() As Boolean
        Property PersonState() As PersonState
        Property Address() As Address
        Property Photo() As Byte()
        Property FamilyPicture() As Byte()
        Overridable Property Phones() As ICollection(Of Phone)
        Overridable Property Companies() As ICollection(Of Company)
    End Class
End Namespace